api-put-object-common.go

```diff
@@ -43,6 +43,19 @@ func isReadAt(reader io.Reader) (ok bool) {
 	return
 }
 
+// isPartUploaded - true if part is already uploaded.
+func isPartUploaded(objPart objectPart, objectParts map[int]objectPart) bool {
```

Comment:
isPartUploaded -  true if part is already uploaded.
This gives an illusion that this part is really uploaded if ret value is true. Instead this function should be more like isPartSelected.